source/tools/themefy.gd

```diff
@@ -84,14 +69,17 @@ func set_icons(theme, dir):
 			if i == "close":
```

Comment:
Well it is gonna be something like:

```
func set_icons(theme, dir):
	for t in theme.get_type_list(""):
		for icon_type in theme.get_icon_list(t):
			var icon = File.new()
			var icon_name = icon_type
			if icon_map.has(icon_type):
				icon_name = icon_map[icon_type]
			var path = find_file(dir, icon_name)
			if "res://" in icon_name:
                                 # since some of them have absolute path
				path = icon_name
			if path:
				icon = load(path)
			theme.set_icon(icon_type, t, icon)

const icon_map = {
	"close": "boxCross",
	"checked": "boxCheckmark",
	"radio_checked": "boxTick",
	"radio_unchecked": "res://interface/grey/grey_circle.png",
	"unchecked": "res://interface/grey/grey_box.png",
	"on": "boxCheckmark",
	"off": "boxCross"
}

```

Reply:
```
>var icon_name = icon_type
>			if icon_map.has(icon_type):
>				icon_name = icon_map[icon_type]
```
This in Godot3.1 would have been much easier with the `get(key, default)` method for `Dictionary`

Reply:
> This is redundant code as well.
> The best thing here actually is having a dictionary that maps all the icon_types with icon name.
> 
> Same applies to the styleboxes above. I might do it in the next minutes

I think I'll open an issue for that, so that we can keep track on it, and maybe I can do it as well so with the issue you will know if I'm working on it already :smile: